### PR TITLE
Update Sphinx docs

### PR DESCRIPTION
**Pull request by accraze**
This PR updates the Sphinx documentation site similar to our other repos and tests the docs build on travis.
It also:
* fixes some RST-related errors in docstrings
* fixes formatting issues in the CHANGELOG
* pins an eslint version due to npm version issues on travis

How to build docs site locally: (w/ python 3)

```
cd docs
make html
cd _build/html
python -m http.server
```
The site should be available at http://localhost:8000

https://phabricator.wikimedia.org/T252173

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ dist: xenial
 os: linux
 
 before_install:
-  - npm install eslint
+  - npm install eslint@6.6.0
```
Comment (halfak) [inline .travis.yml:8]: I just got stuck on this yesterday for a PR in Wikilabels.  I came to the same solution.

File: docs/conf.py [hunk 2]
```diff
@@ -49,18 +76,18 @@
...
 # The version info for the project you're documenting, acts as replacement for
 # |version| and |release|, also used in various other places throughout the
 # built documents.
 #
 # The short X.Y version.
-version = '0.1.0'
+version = ores.__version__
 # The full version, including alpha/beta/rc tags.
-release = '0.1.0'
+release = ores.__version__
```
Comment (halfak) [inline docs/conf.py:90]: Nice

File: ores/scoring_context.py [hunk 3]
```diff
@@ -233,11 +233,14 @@ def from_config(cls, config, name, section_key="scoring_contexts"):
                     scorer_models:
                         damaging: enwiki_damaging_2014
                         good-faith: enwiki_good-faith_2014
+
                     extractor: enwiki
+
```
Comment (halfak) [inline ores/scoring_context.py:238]: No concerns really but why the added space?

File: ores/wsgi/util.py [hunk 4]
```diff
@@ -67,11 +67,11 @@ def format_error(error):
 def build_score_request(scoring_system, request, context_name=None, rev_id=None,
                         model_name=None):
     """
-    Build an :class:`ores.ScoreRequest` from information contained in a
+    Build an :class:`ores.score_request.ScoreRequest` from information contained in a
     request.
 
     :Parameters:
-        scoring_system : :class:`ores.ScoringSystem`
+        scoring_system : :class:`ores.scoring_systems.scoring_system.ScoringSystem`
```
Comment (halfak) [inline ores/wsgi/util.py:74]: I think we want to reference: https://github.com/wikimedia/ores/blob/master/ores/scoring_systems/__init__.py#L3

So `ores.scoring_systems.ScoringSystem`.

**Comment by halfak** [comment]
Really just one note.  Nice work otherwise.  It'll be nice to have docs build with CI.

File: ores/scoring_context.py (hunk 3 above)
Comment (accraze) [reply ores/scoring_context.py:238]: The space is required by the RST spec to denote the end of a 'list' and was showing up as a warning during the sphinx-build test:
```
/ores/scoring_context.py:docstring of ores.scoring_context.ScoringContext.from_config:8:
Definition list ends without a blank line; unexpected unindent.
```

**Pull request merged**